changelog_gen/extractor.py

```diff
@@ -82,13 +82,22 @@ def extract_version_tag(sections: SectionDict) -> str:
     Bugs/Fixes: patch
 
     """
+    version_info_ = BumpVersion.get_version_info("patch")
+    current = version_info_["current"]
     # TODO(edgy): In the case of 0.x releases, everything should be one semver lower.
     # https://github.com/EdgyEdgemond/changelog-gen/issues/50
+    semvers = ["patch", "minor", "major"]
     semver = "minor" if "feat" in sections else "patch"
     for section_issues in sections.values():
         for issue in section_issues.values():
             if issue["breaking"]:
                 semver = "major"
+
+    if current.startswith("0."):
+        # If currently on 0.X releases, downgrade semver by one, major -> minor etc.
+        idx = semvers.index(semver)
+        semver = semvers[max(idx - 1, 0)]
+
```

Comment:
🔧  would it be simpler to deal with index the whole time?
```
semvers = ["patch", "minor", "major"]
ver_idx = 0  # start as patch
if "feat" in sections:
  ver_idx = 1 

for section_issues in sections.values():
  for issue in section_issues.values():
    if issue["breaking"]:
      ver_idx = len(semvers) - 1

if current.startwith("0."):
  ver_idx = max(ver_idx - 1, 0)

semver = semvers[ver_idx]
```

Reply:
this gets rearranged a little in a later PR, with simplfication of configuration keys. Will take a look at this in the context of that change.

Reply:
the later change makes this idea less clean, as we lose the ability to do `if feat in sections` partly due to using dataclasses, and due to support for custom sections, minor could be `feature`, or multiple types.

Reply:
https://github.com/EdgyEdgemond/changelog-gen/pull/72/files#diff-5ec09b87a087c81907fc97531029c0a529055bc916aa368dacc36cfb429fe67aR118